src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -88,23 +91,35 @@ protected OtpUser getUserProfile(Auth0UserProfile profile) {
      * HTTP endpoint to send an SMS text to an {@link OtpUser}'s phone number with a verification code. This is used
      * during user signup (or if a user wishes to change their notification preferences to use a new un-verified phone
      * number).
+     * Before sending a SMS request, this endpoint also saves the submitted phone number, marks it as not verified.
      */
     public VerificationResult sendVerificationText(Request req, Response res) {
         OtpUser otpUser = getEntityForId(req, res);
-        if (otpUser.phoneNumber == null) {
-            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "User must have valid phone number to verify.");
+        // Get phone number from the path param.
+        String phoneNumber = req.params(PHONE_PARAM);
+        if (StringUtils.isBlank(phoneNumber)) {
+            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "A phone number must be provided.");
         }
-        Verification verification = NotificationUtils.sendVerificationText(otpUser.phoneNumber);
+
+        // Update OtpUser before submitting SMS request.
+        otpUser.phoneNumber = phoneNumber;
+        otpUser.isPhoneNumberVerified = false;
+        Persistence.otpUsers.replace(otpUser.id, otpUser);
+/*
+        Verification verification = NotificationUtils.sendVerificationText(phoneNumber);
         if (verification == null) {
             logMessageAndHalt(req, HttpStatus.INTERNAL_SERVER_ERROR_500, "Unknown error sending verification text");
         }
         // Verification result will show "pending" status if verification text is successfully sent.
         return new VerificationResult(verification);
+ */
+        return null;
```

Comment:
I'm not sure what's happening here... returning null doesn't seem too useful in this scenario! I'm guessing this is just a temporary comment that accidentally got checked in?

Reply:
Yes... thanks for catching that.